fetch.bs

```diff
@@ -2353,7 +2435,374 @@ X-Content-Type-Options           = "nosniff" ; case-insensitive</pre>
 pertain to them. Also, considering "<code>image</code>" was not compatible with deployed content.
 
+<h3 id=token-binding>Token Binding</h3>
+
+<p>In order to protect security tokens like HTTP cookies and OAuth tokens, user agents and servers
+can use a technique known as <dfn export id=concept-token-binding>Token Binding</dfn> to
+cryptographically associate a given token with a secret
+known only to a specific user agent
+(a <dfn export id=concept-token-binding-key>token-binding key</dfn>).
+The user agent uses a different <a for=/>token-binding key</a>
+for every different <a for=/>registrable domain</a>.
+The association of key and token mitigates the risk that attackers
+can steal the token and use it themselves, as they will not be able to
+easily replicate the user agent's secret,
+and therefore will be unable to replicate the cryptographic binding of the token.
+
+<p>Details are described in [[TOKBIND-NEGOTIATION]], [[TOKBIND-PROTOCOL]] and
+[[TOKBIND-HTTPS]] and integration is defined here.
+
+<p>A <dfn export id=token-binding-id>token binding ID</dfn> is the non-secret representation
+of a <a for=/>token-binding key</a>, as described in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.2">section 3.2</a>
+of [[TOKBIND-PROTOCOL]].
+
+<p>At a very high level, a user agent negotiates the use of Token Binding with
+a server when it sets up a TLS connection to the server,
+and then uses Token Binding for requests that it sends over that connection.
+
+<p class="note no-backref">
+The user agent can use <a for=/>Token Binding</a> for any
+<a for=/>request</a> sent over a TLS connection for which Token Binding
+negotiation was successful.
+
+<p class="note no-backref">
+Token Binding works correctly even when requests for different origins are
+coalesced on the same HTTP/2 connection.  The user agent will use
+a different <a for=/>token-binding key</a> for every different
+<a for=/>registrable domain</a>.
+Separate Token Binding messages (generated using different token-binding
+keys) will be sent as headers in the requests to the different origins.
+This implies that all origins that can be coalesced into the same
+HTTP/2 connection need to support the same Token Binding protocol
+version and key parameters.
+
+<h4 id=negotiating-token-binding>Negotiating Token Binding</h4>
+
+The user agent proposes and agrees to the use of Token Binding as described in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-negotiation#section-2">section 2</a>
+of the Token Binding Negotiation spec [[!TOKBIND-NEGOTIATION]].
+While setting up a TLS <a>connection</a>, it sends its highest supported
+<dfn export id=concept-token-binding-protocol-version for=connection>token-binding protocol version</dfn>
+and all supported cryptographic algorithms and parameters (the
+<dfn export id=concept-token-binding-key-parameters for=connection>token-binding key parameters</dfn>),
+in order of preference,
```

Comment:
> in order of preference,

If we want to make it possible for a server to try to migrate users from one key type to another, this probably needs to be explained in a lot more detail what "order of preference" means for a client. It probably depends on what keys the client already has in its key store for the domain it's trying to connect to. (There are more complications than just that here.)